app/catalog/catalog.jsx

```diff
@@ -16,21 +332,60 @@ export default function Catalog() {
         await setItems(result.items);
       } catch (error) {
         console.log(error.message);
-      } 
+      }
     }
     getItems();
   }, []);
 
-  const displayedItems = items.map((item, idx) => {
-    return (<div key={idx}>
-      <h2>{item.name}</h2>
-      <div>{item.price / 100}</div>
-    </div>)
+  const toggleAvailability = (idx) => {
+    refs[idx].current.className = (refs[idx].current.className === "hidden") ? "" : "hidden";
+  }
+
+  const addToCart = (e, idx, item) => {
+    e.preventDefault();
+    const [inputStart, inputEnd] = refs[idx].current.children[0].getElementsByTagName("input");
+    const endDate = new Date (inputEnd.value.slice(0, 4), inputEnd.value.slice(5, 7) - 1, inputEnd.value.slice(8, 10));
+    const startDate = new Date (inputStart.value.slice(0, 4), inputStart.value.slice(5, 7) - 1, inputStart.value.slice(8, 10));
+    const numOfReservedDays = (endDate - startDate === 0) ? 1 : ((endDate - startDate) / 86400000);
```

Comment:
we should make the magic number 86400000 it's own variable to help make the code understandable to others